examples/dynamic_rendering/source/dynamic_rendering.cpp

```diff
@@ -0,0 +1,244 @@
+				ImGui::Checkbox("Fullscreen viewport", &mFullscreenViewport);
+				ImGui::End();
+			});
+		}
+	}
+
+	void update() override
+	{
+		// On Esc pressed,
+		if (avk::input().key_pressed(avk::key_code::escape) || avk::context().main_window()->should_be_closed()) {
+			// stop the current composition:
+			avk::current_composition()->stop();
+		}
+	}
+
+	void render() override
+	{
+		auto mainWnd = avk::context().main_window();
+		auto inFlightIndex = mainWnd->in_flight_index_for_frame();
+
+		// Get a command pool to allocate command buffers from:
+		auto& commandPool = avk::context().get_command_pool_for_single_use_command_buffers(*mQueue);
+
+		// The swap chain provides us with an "image available semaphore" for the current frame.
+		// Only after the swapchain image has become available, we may start rendering into it.
+		auto imageAvailableSemaphore = mainWnd->consume_current_image_available_semaphore();
+		
+		// Create a single command buffer:
+		auto cmdBfr = commandPool->alloc_command_buffers(1u, vk::CommandBufferUsageFlagBits::eOneTimeSubmit);
+		
+		avk::image_view swapchainImageView = mainWnd->current_image_view();
+		avk::image_t const & swapchainImage = mainWnd->current_image_reference(); 
+		avk::attachment colorAttachment = avk::attachment::declare_for(swapchainImageView.as_reference(), avk::on_load::clear, avk::usage::color(0), avk::on_store::store);
+
+		std::vector<avk::recorded_commands_t> render_commands = {};
+		// First we transition the swapchain image into color attachment format
+		render_commands.emplace_back(
+			avk::sync::image_memory_barrier(swapchainImage,
+				// source stage none because it is handled by imageAvailable semaphore
+				avk::stage::none >> avk::stage::color_attachment_output,
+				avk::access::none >> avk::access::color_attachment_write
+				// We don't care about the previous contents of the swapchain image (they probably not even defined!)
+			).with_layout_transition(avk::layout::undefined >> avk::layout::color_attachment_optimal)
+		);
+
+		auto const windowResolution = mainWnd->resolution();
+		// Set the dynamic viewport only once at the start if we want shared fullscreen viewport for each renderpass
+		if(mFullscreenViewport) {
+			auto const currentViewport = vk::Viewport(0.0f, 0.0f, windowResolution.x, windowResolution.y);
+			render_commands.emplace_back(avk::command::custom_commands([=](avk::command_buffer_t& cb) {
+				cb.handle().setViewport(0, 1, &currentViewport);
+			}));
+		} 
+
+		// We render the image in four renderpasses - each renderpass has it's own clear value as well as offset and extent.
+		// Note that the renderpass extent acts as "clip" on top of the pipeline viewport (which is set to be the entire window).
+		// Because of this we still get a single triangle just drawn in four parts, if we instead wanted to see four triangles we would
+		// need to set the pipeline viewport to be 
+		for(uint32_t renderpassIdx = 0; renderpassIdx < 4; renderpassIdx++)
```

Comment:
Isn't it a bit confusing to call this "renderpass"? Because a renderpass is something very specific in Vulkan.

Reply:
I didn't really think about that, you are right it is confusing. What name do you think would be better? I came up with either "dynamicRenderpass" or maybe better "renderingScope" but I still am not fully happy with these... Do you have any ideas?

Reply:
...especially given the fact that dynamic rendering does not even use a `vk::RenderPass`.

Reply:
Well, actually it is just 4 draw calls, right? So why not call it `draw`?

Reply:
Good idea, I'll call them draws. 👍